mantle/platform/api/aliyun/api.go

```diff
@@ -134,6 +134,20 @@ func (a *API) CopyImage(source_id, dest_name, dest_region, dest_description, kms
 			Value: "mantle",
 		},
 	}
+
+	// Check if an image with the name has already been uploaded. This can
+	// happen when replication to a region fails which happens often on aliyun
+	images, err := a.GetImagesInRegion(dest_name, dest_region)
+	if err != nil {
+		return "", fmt.Errorf("getting image: %v", err)
+	}
+
+	// return early if there is already an image with that tag
+	if len(images.Images.Image) > 0 {
+		plog.Infof("image with name %v in %v region already exists--skipping copy", dest_name, dest_region)
+		return images.Images.Image[0].ImageId, nil
```

Comment:
i wonder if we should somehow respect `wait_for_ready` here too? I guess it's possible that the image pre-exists but isn't ready yet.